projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad.php

```diff
@@ -680,6 +680,33 @@ function wpcom_launchpad_set_fse_next_steps_modal_hidden( $should_hide ) {
 	return update_option( 'wpcom_launchpad_config', $wpcom_launchpad_config );
 }
 
+/**
+ * Helper function to set the current active checklist in the navigator context.
+ *
+ * @param string $checklist_slug The slug of the launchpad task list to mark as active.
+ * @return bool Whether the option update succeeded.
+ */
+function wpcom_launchpad_set_current_active_checklist( $checklist_slug ) {
+	$wpcom_launchpad_config = get_option( 'wpcom_launchpad_config' );
+
+	$checklists = wpcom_launchpad_checklists()->get_all_task_lists();
+	if ( ! array_key_exists( $checklist_slug, $checklists ) ) {
+		return false;
+	}
+
+	if ( ! $wpcom_launchpad_config || ! is_array( $wpcom_launchpad_config ) ) {
+		$wpcom_launchpad_config = array();
+	}
+
+	if ( isset( $wpcom_launchpad_config['active_checklist_slug'] ) && $checklist_slug === $wpcom_launchpad_config['active_checklist_slug'] ) {
+		return true;
+	}
+
+	$wpcom_launchpad_config['active_checklist_slug'] = $checklist_slug;
```

Comment:
We should handle `null` here as well.
```suggestion
	if ( null === $checklist_slug ) {
		if ( ! isset( $wpcom_launchpad_config['active_checklist_slug'] ) ) {
			return true;
		}
		unset( $wpcom_launchpad_config['active_checklist_slug'] );
	} else {
		if ( isset( $wpcom_launchpad_config['active_checklist_slug'] ) && $checklist_slug === $wpcom_launchpad_config['active_checklist_slug'] ) {
			return true;
		}
		$wpcom_launchpad_config['active_checklist_slug'] = $checklist_slug;
	}
```